### PR TITLE
Convert Syndrome sample to standalone executable to enable QIR testing

### PR DESCRIPTION
**Pull request by samarsha**
Replaced the Python driver with a Q# entry point for the Syndrome error correction sample,  so that the QIR validation script in the QDK repo can run it.

### CODE REVIEW

File: samples/error-correction/syndrome/Syndrome.qs [hunk 1]
```diff
@@ -2,13 +2,36 @@
...
+    /// # Summary
+    /// Runs the Syndrome sample.
+    ///
+    /// # Input
+    /// ## qubits
+    /// The number of qubits to use.
+    @EntryPoint()
+    operation RunSyndrome(qubits : Int) : Unit {
+        let qubitIndices = Shuffle(RangeAsIntArray(0 .. qubits - 1));
+        let inputValues = DrawMany(DrawRandomBool, qubits, 0.5);
```
Comment (cgranade) [inline samples/error-correction/syndrome/Syndrome.qs:23]: Same comment here, this is great but may benefit from some additional comments to help new users understand what the results of these two lines will be.

**Review by cgranade — APPROVED** [review]

**Review by kuzminrobin — APPROVED** [review]

File: samples/error-correction/syndrome/Syndrome.qs [hunk 2]
```diff
@@ -99,4 +122,38 @@ namespace Microsoft.Quantum.Samples.ErrorCorrection.Syndrome {
...
+    /// # Summary
+    /// Chooses a random element from a non-empty array. Fails if the array is empty.
+    ///
+    /// # Input
+    /// ## xs
+    /// The array.
+    ///
+    /// # Output
+    /// A random element from the array.
+    internal operation Choose<'a>(xs : 'a[]) : 'a {
```
[the remark this answers is not in the document]
Comment (samarsha) [reply samples/error-correction/syndrome/Syndrome.qs:154]: My only concern is that it's a partial function... I wish we could do something like

```qsharp
newtype NonEmpty<'a> = (Head : 'a, Tail : 'a[]);
```

File: samples/error-correction/syndrome/Syndrome.qs [hunk 3]
```diff
@@ -99,4 +122,38 @@ namespace Microsoft.Quantum.Samples.ErrorCorrection.Syndrome {
...
+    /// # Summary
+    /// Shuffles the order of elements in an array.
+    ///
+    /// # Input
+    /// ## xs
+    /// The array.
+    ///
+    /// # Output
+    /// The shuffled array.
+    internal operation Shuffle<'a>(xs : 'a[]) : 'a[] {
```
[the remark this answers is not in the document]
Comment (samarsha) [reply samples/error-correction/syndrome/Syndrome.qs]: Do you want to keep track of them as ideas for library additions anywhere?

File: samples/error-correction/syndrome/Syndrome.qs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgranade) [reply samples/error-correction/syndrome/Syndrome.qs]: Fair, yeah. Even in absentia `NonEmpty`, something like a proper `Maybe<'T>` along with a partial function that coerces to `'T` could help make `MaybeChooseElement` a bit easier to use.

File: samples/error-correction/syndrome/Syndrome.qs (hunk 3 above)
[the remark this answers is not in the document]
Comment (cgranade) [reply samples/error-correction/syndrome/Syndrome.qs]: If you'd be willing to file a proposal as an issue on QuantumLibraries, I'd greatly appreciate it!

**Pull request merged**